unit/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -505,26 +505,6 @@ TEST_CASE(
         test.sent_commands ==
         std::vector<smt_commandt>{smt_get_value_commandt{foo_term}});
     }
-    SECTION("Get value of non-set symbol")
-    {
-      // smt2_incremental_decision_proceduret is used this way when cbmc is
-      // invoked with the combination of `--trace` and `--slice-formula`.
-      test.sent_commands.clear();
-      const exprt bar =
-        make_test_symbol("bar", signedbv_typet{16}).symbol_expr();
-      REQUIRE(test.procedure.get(bar) == bar);
-      REQUIRE(test.sent_commands.empty());
-    }
-    SECTION("Get value of type less symbol back")
-    {
-      // smt2_incremental_decision_proceduret is used this way as part of
-      // building the goto trace, to get the partial order concurrency clock
-      // values.
-      test.sent_commands.clear();
-      const symbol_exprt baz = symbol_exprt::typeless("baz");
```

Comment:
Yes, but the values of those symbols will be known (must be known!) to the back-end.

Reply:
I think we are currently missing unit tests where the value of a symbol is known, but type less. Although I guess this would fail at the point of sending to the solver anyway as I am unsure how the value of a type less symbol should be encoded into SMT terms. I don't want to hold up this PR over the lack of a test for this. I also don't really see why it should be valid for a symbol to be type less by the time it reaches the decision procedure.

Reply:
I believe the symbol won't be lacking a type as far as the decision procedure is concerned, just the earlier stage requesting the symbol value will not know what exact type it is.